src/core/execute.c

```diff
@@ -3870,12 +3870,25 @@ static int exec_child(
 #endif
 
                 /* PR_GET_SECUREBITS is not privileged, while PR_SET_SECUREBITS is. So to suppress potential EPERMs
-                 * we'll try not to call PR_SET_SECUREBITS unless necessary. */
-                if (prctl(PR_GET_SECUREBITS) != secure_bits)
+                 * we'll try not to call PR_SET_SECUREBITS unless necessary. Setting securebits requires
+                 * CAP_SETPCAP. */
+                if (prctl(PR_GET_SECUREBITS) != secure_bits) {
+                        /* CAP_SETPCAP is required to set securebits. This capabilitiy is raised into the
+                         * effective set here. The effective set is overwritten with the values
+                         * from the
+                         * - ambient set during execve if executed by non-root
+                         * - (inheritable | bounding) set during execve if executed by root
+                         */
```

Comment:
given that it is easy, wouldn't it make sense to just reset the old caps after we did the PRT_SET_SECUREBITS thing?

Reply:
I am not sure. 
This might create the impression that capabilities have to be unset here, although the kernel resets them during execve.
Hence imho this would be code duplication.